src/goto-symex/goto_state.cpp

```diff
@@ -86,7 +89,10 @@ void goto_statet::apply_condition(
         increase_generation(
           l1_identifier, l1_lhs, previous_state.get_l2_name_provider());
 
-        propagation[l1_identifier] = rhs;
+        if(propagation.has_key(l1_identifier))
```

Comment:
Might want to just give `sharing_mapt` a primitive `set` operation to save the double lookup

Reply:
@danpoe maybe?

Reply:
I'd recommend to use `find(l1_identifier)` here and then check that `rhs` is different from the current value in the map, and only then do the replacement. When using this idiom a separate `set()` method should be unnecessary.

Reply:
Done.